lib/Controller/SAMLController.php

```diff
@@ -157,6 +161,27 @@ private function autoprovisionIfPossible() {
 		}
 	}
 
+	/**
+	 * @throws UserFilterViolationException
+	 */
+	protected function assertGroupMemberships(): void {
+		$groups = $this->userData->getGroups();
+		$settings = $this->samlSettings->get($this->session->get('user_saml.Idp'));
+
+		$rejectGroupsString = $settings['saml-user-filter-reject_groups'] ?? '';
+		$rejectGroups = array_map('trim', explode(',', $rejectGroupsString));
+
+		if (!empty(array_intersect($groups, $rejectGroups))) {
+			throw new UserFilterViolationException('User is member of a rejection group.');
+		}
+
+		$requireGroupsString = trim($settings['saml-user-filter-require_groups'] ?? '');
+		$requireGroups = array_map('trim', explode(',', $requireGroupsString));
```

Comment:
It is not possible to use json because the UI is entering a comma separated group list, right?
At least avoid trimming twice.

```suggestion
		$requireGroupsString = $settings['saml-user-filter-require_groups'] ?? '';
		$requireGroups = array_map('trim', explode(',', $requireGroupsString));
```

Reply:
> It is not possible to use json because the UI is entering a comma separated group list, right?

that's correct. This trim is later for the `empty` check.